File: back_end/user-service/src/main/java/org/example/userservice/service/UserService.java
package org.example.userservice.service;

import org.example.userservice.dto.UserDTO;
import org.example.userservice.dto.WatchHistoryDTO;
import org.example.userservice.dto.WatchlistDTO;

import java.util.List;
import java.util.Map;

public interface UserService {
    UserDTO createUser(UserDTO userDTO);

    UserDTO getUserById(Long id);

    List<UserDTO> getAllUsers();

    // Watchlist
    WatchlistDTO addVideoToWatchlist(Long userId, Long videoId);

    void removeVideoFromWatchlist(Long userId, Long videoId);

    List<WatchlistDTO> getUserWatchlist(Long userId);

    // Watch History
    WatchHistoryDTO recordWatchHistory(Long userId, Long videoId, Integer progressTime, Boolean completed);

    List<WatchHistoryDTO> getUserWatchHistory(Long userId);

    // Statistics
    Map<String, Object> getUserViewingStatistics(Long userId);
}

File: front_end/src/services/userService.js
import api from './api';

export const userService = {
    getUserById: async (id) => {
        const response = await api.get(`/users/${id}`);
        return response.data;
    },

    getAllUsers: async () => {
        const response = await api.get('/users');
        return response.data;
    },

    getUserStatistics: async (id) => {
        const response = await api.get(`/users/${id}/statistics`);
        return response.data;
    }
};
